circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -377,7 +416,17 @@ def _get_mapping_ids_by_partition(
         subcirc_map_ids.append([])
         for i, inst in enumerate(circ.data):
             if isinstance(inst.operation, SingleQubitQPDGate):
-                decomp_id = int(inst.operation.label.split("_")[-1])
+                try:
+                    decomp_id = int(inst.operation.label.split("_")[-1])
+                except (AttributeError, ValueError):
+                    raise ValueError(
+                        "SingleQubitQPDGate instances in input circuit(s) must have their "
+                        'labels suffixed with "_<id>", where <id> is the index of the cut '
+                        "relative to the other cuts in the circuit. For example, all "
+                        "SingleQubitQPDGates belonging to the same cut, N, should have labels "
```

Comment:
Since this is an example (and not a strict requirement), I prefer "could" rather than "should" here.  Or perhaps there's an even better way to phrase it. ... like "One way to satisfy this is to have ..."

```suggestion
                        "SingleQubitQPDGates belonging to the same cut, N, could have labels "
```
(will also require an update to the test)

Reply:
I'm not sure I understand? Having an integer appended to the end of the label is required here

Reply:
There is an implied "[If using this function], SingleQubitQPDGates *should* have labels ..."

Reply:
Maybe you are right though...maybe the integer requirement is too restrictive. We could handle strings and use dictionaries....

Reply:
sorry, the relevant part of this sentence (whose meaning would change with "could") is actually on the next line.  It _needs_ to have the suffix, but it doesn't _need_ to be `formatted as "<gate_label>_N"` (it could be anything, but it's the gate label if you use the other parts of our workflow).

Reply:
Ohhhh I see. I was using `gate_label` generically, but it actually has a specific meaning.

Maybe we can leave "should" and change this to `<your_label>_N`?

Reply:
> Maybe we can leave "should" and change this to <your_label>_N?

Yeah, that sounds great.